Makefile

```diff
@@ -75,7 +75,7 @@ ifneq ($(USE_BLAS), mkl)
 ifeq ($(MKL_EXTERNAL), 1)
 	MKL_LDFLAGS+=-Wl,-rpath,$(MKLROOT)/lib
 	CFLAGS += -I$(MKLROOT)/include
-	LDFLAGS += -L$(MKLROOT)/lib/ -liomp5 -lmklml_gnu
+	LDFLAGS += -Wl,--as-needed -L$(MKLROOT)/lib/ -liomp5 -lmklml_intel
```

Comment:
To verify this, you can try to compile and use "ldd lib/libmxnet.so |grep omp" to check the link dependency. W/o this, linker will link both gomp and iomp which cause problem. With this flag, linker will only link to iomp.